File: app/src/main/java/com/fishweeb/practical/Collidable.java
package com.fishweeb.practical;

public interface Collidable
{
    String GetType();

    float GetPosX();
    float GetPosY();

    float GetRadius();
    float GetWidth();
    float GetHeight();

    void OnHit(Collidable _other);

    float GetDamage();
    boolean GetDead();
}

File: app/src/main/kotlin/com/fishweeb/practical/Collidable.kt
package com.fishweeb.practical

interface Collidable {
    fun GetType(): String?

    fun GetPosX(): Float
    fun GetPosY(): Float

    fun GetRadius(): Float
    fun GetWidth(): Float
    fun GetHeight(): Float

    fun OnHit(_other: Collidable?)

    fun GetDamage(): Float
    fun GetDead(): Boolean
}
